src/Kmd.Logic.Gateway.Automation/Client/ValidatePublishingRequest.cs

```diff
@@ -17,5 +17,16 @@ public ValidatePublishingRequest(Guid providerId, IEnumerable<ApiValidationModel
         public IEnumerable<ApiValidationModel> Apis { get; }
 
         public IEnumerable<ProductValidationModel> Products { get; }
+
+        public void Dispose()
+        {
+            if (this.Apis != null)
+            {
+                foreach (var api in this.Apis)
+                {
+                    api.OpenApiSpec?.Dispose();
```

Comment:
I believe that they are disposed once they are sent with http call

Reply:
No. We were getting "file already open" error in the api publishing code when it tried to open the file for reading. Then we realized the stream opened by the validator was not getting closed. Hence doing it explicitly

Reply:
So how is it possible that on calling ValidatePublishing I didn't have to dispose them?

Reply:
Ok, I got it right now. But test then if validation call still works ;)